### DISCUSSION TITLE
Control the sketcher using JavaScript

### DISCUSSION THREAD

**Discussion by BenClaveau** [epam/ketcher#4050] [Ideas]
I am working on a retrosynthesis tool that uses ``ketcher-react`` (2.15.1). In this tool, I want to select atoms or bonds to indicate how to break the molecule. Currently, I am using RDKit.js to highlight atoms or bonds. I would like to be able to perform this action directly in the sketcher, but I don't know how to do it. 
Ideally, I would like to add a button to select one or more atoms and bonds and retrieve this list in my React code. MarvinJS allows this to be done via a right-click.
<img width="426" alt="Capture d’écran 2024-02-07 à 15 01 12" src="https://github.com/epam/ketcher/assets/5556343/4e5ecd66-efc0-41af-ae3e-93bc9ef19bf4">
Thanks

**Comment by vanoprenko** [comment]
Highlighting atoms and bonds is already supported in Ketcher API. You can do this as simple as this:
`ketcher.editor.highlights.create( {atoms: [152, 156], bonds: [121,122], color: 'red'})`

We also are going to introduce the ability to do this from Ketcher UI via context menu soon.

**Reply by BenClaveau** [reply]
Thank you @vanoprenko, Is it possible to add a button to facilitate interaction? When the button is pressed, each time an atom is clicked, it will be selected?
